contrib/src/main/java/com/datatorrent/contrib/kafka/AbstractKafkaInputOperator.java

```diff
@@ -392,9 +412,20 @@ public void emitTuples()
     if (maxTuplesPerWindow > 0) {
       count = Math.min(count, maxTuplesPerWindow - emitCount);
     }
-    for (int i = 0; i < count; i++) {
+    if (count > 0) {
+      // If the total size transmitted in the window will be exceeded don't transmit anymore messages in this window
+      // Make an exception for the case when no message has been transmitted in the window and transmit at least one
+      // message even if the message size is greater than max message size so that the processing doesn't get stuck
+      if ((emitCount > 0) && ((maxTotalMsgSizePerWindow - emitTotalMsgSize) < consumer.peekMessage().msg.size())) {
```

Comment:
I had thought about that but then you can have more than queue size number of outstanding messages, you would have queue size + 1 which would violate the limit.